app/controllers/api/v1/links_controller.rb

```diff
@@ -26,6 +26,16 @@ def show
     end
   end
 
+  def top_links
+    query = Link.order(click_count: :desc).limit(5)
+
+    query = query.joins(:tags).where(tags: { name: params[:tag] }) if params[:tag].present?
+
+    links = query.distinct
+
```

Comment:
do you need a .distinct here? I'm trying to think of a time the query would return a query with duplicates